evm_arithmetization/src/cpu/kernel/tests/rlp/encode.rs

```diff
@@ -39,7 +43,11 @@ fn test_encode_rlp_scalar_medium() -> Result<()> {
 
     interpreter.run()?;
     let expected_stack = vec![pos + U256::from(4)]; // pos' = pos + rlp_len = 2 + 4
-    let expected_rlp = vec![0, 0, 0x80 + 3, 0x01, 0x23, 0x45];
+
+    // The two first values of the RLP segment are trie_data_len (here 0), and the
+    // hardcoded 0x80 for an empty node.
+    let expected_rlp = vec![0, 0x80, 0x80 + 3, 0x01, 0x23, 0x45];
```

Comment:
Is `trie_data_len` supposed to be 0? (Is it updated anywhere? Or is it just the length of the initial MPTs?). I am not sure I understand the need for `trie_data_len` actually, since we have the global metadata for the size of the segment. Am I missing something?

Reply:
We initially store at offset 0 the length of `TrieData` when initializing the RLP segment. But for this test (and in others), it doesn't matter. As we update address at offset 1 for the empty node, it is initialized with 0.

Reply:
Sorry, I'm still a bit confused about trie_data_len: I'm not sure where we need it?

Reply:
Hmm, I think I may have been confused by this comment in the end:

```rust
    /// The size of the `TrieData` segment, in bytes, represented as a whole
    /// address. In other words, the next address available for appending
    /// additional trie data.
    RlpDataSize,
```

but it's not really accurate, and we store `SEGMENT_RLP_RAW` in `main.asm` (i.e. the initial address). So mea culpa, I'll update it tomorrow (and I'll fix this comment).

Reply:
It's updated. Now EMPTY_NODE_ADDR is at offset 0, and we initialize the next RLP address in main.asm as INITIAL_TXN_RLP_ADDR (which is at offset 1).